src/metabase/db/query.clj

```diff
@@ -42,9 +42,13 @@
   (^String [^String sql db-type]
    (when sql
      (let [formatter (SqlFormatter/of (case db-type
-                                        :mysql    Dialect/MySql
-                                        :postgres Dialect/PostgreSql
-                                        :h2       Dialect/StandardSql))]
+                                        :mysql     Dialect/MySql
```

Comment:
I made some tweaks to the SQL formatting in errors and test context. Since this used more generally now I added mappings for other database types we support outside of the "big 3" application database types.